### ISSUE TITLE
User abuse reports - search results should be returned when searching by the user name or user id 

### ISSUE THREAD

**Issue by ioanarusiczki** [mozilla/addons-server#11308]
Followup for https://github.com/mozilla/addons-server/issues/11133

STR:
1. Go on dev to /en-US/admin/models/abuse/abusereport/
2. Filter the reports by type: users
3. Try to see the reports made for one user only by typing in the user's name (or user's id)

Expected:
Reports for the specified user are listed.

Actual result:
No results found. 

Notes:
For an addon the abuse reports are displayed when searching by the addon's name, slug or guid.
Reproducible on admin tool dev - FF66(Win10). 

![search using a users name](https://user-images.githubusercontent.com/33448286/57083373-1ec2f980-6d01-11e9-8175-c56ec3b77b8f.gif)

**Comment by diox** [comment]
Yeah, I should be smarter about this. It should totally depend on the type selected, there is no need to search through add-on fields if users type is selected, and vice-versa.

**Comment by ioanarusiczki** [comment]
@diox Here are my test results on django admin dev - FF67(Win10). 
- searching by id or GUID works for both users and addons, I did not find problems.
- searching by Name for Users doesn't really work, at least not with most of the Names I tested with. In many cases, if I type in the complete names I get 0 search results.
I'll give examples for this situation:
BobsDisplayName 
UserWithNoPerms 
LexaDev
IoanaTestingStageandDevlongstringlong
Itsy Bitsy Spider
Amoga-dev 
Jorge Villalobos 

These are some examples of Names that returned the expected result:
CosminSV
kumar_dev2 
testemaildev 

- searching by Name addons works well in most of the cases. Still, if a name is longer it will return less results than it should.
Examples 
"X-notifier (for Gmail,Hotmail,Yahoo,AOL ...)" ->130 search results using the id or GUID then 13 using it's complete Name.
"Awesome Screenshot Plus - Capture, Annotate & More"-> 122 search results using the id down to 61 when I try with the complete Name.
"Ecosia - The search engine that plants trees!" -> I get 5 results if I type in just "Ecosia" (or search it by GUID or ID), and 1 result if I use it's complete Name.
